### ISSUE TITLE
Python 3.7 tests are broken

### ISSUE THREAD

**Issue by jonringer** [jmcgeheeiv/pyfakefs#508]
**Describe the bug**
Packaging this for nixpkgs, and noticed the python3.7 test suite is broken with the following error:

<details> <summary> Error log </summary>

```
======================================================================
ERROR: test_glob (pyfakefs.tests.fake_pathlib_test.FakePathlibPathFileOperationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pyfakefs-3.7/pyfakefs/tests/fake_pathlib_test.py", line 710, in test_glob
    self.assertEqual(sorted(path.glob('*.py')),
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 1119, in glob
    selector = _make_selector(tuple(pattern_parts), self._flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 464, in _make_selector
    return cls(pat, child_parts, flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 520, in __init__
    self.match = flavour.compile_pattern(pat)
AttributeError: '_FakePosixFlavour' object has no attribute 'compile_pattern'

======================================================================
ERROR: test_glob_case_posix (pyfakefs.tests.fake_pathlib_test.FakePathlibPathFileOperationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pyfakefs-3.7/pyfakefs/tests/fake_pathlib_test.py", line 733, in test_glob_case_posix
    self.assertEqual(sorted(path.glob('*.py')),
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 1119, in glob
    selector = _make_selector(tuple(pattern_parts), self._flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 464, in _make_selector
    return cls(pat, child_parts, flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 520, in __init__
    self.match = flavour.compile_pattern(pat)
AttributeError: '_FakePosixFlavour' object has no attribute 'compile_pattern'

======================================================================
ERROR: test_glob_case_windows (pyfakefs.tests.fake_pathlib_test.FakePathlibPathFileOperationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/pyfakefs-3.7/pyfakefs/tests/fake_pathlib_test.py", line 721, in test_glob_case_windows
    self.assertEqual(sorted(path.glob('*.py')),
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 1119, in glob
    selector = _make_selector(tuple(pattern_parts), self._flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 464, in _make_selector
    return cls(pat, child_parts, flavour)
  File "/nix/store/v543lrn8xkx5ylsa6g59l8s74jx3nh6p-python3-3.7.6/lib/python3.7/pathlib.py", line 520, in __init__
    self.match = flavour.compile_pattern(pat)
AttributeError: '_FakePosixFlavour' object has no attribute 'compile_pattern'

----------------------------------------------------------------------
Ran 1901 tests in 4.445s

FAILED (errors=3, skipped=745, expected failures=2)
```
</details>

It's also only failing on 3.7. 2.7,3.5,3.6,3.8 all seem to pass fine.

**How To Reproduce**
```
python3.7 -m pyfakefs.tests.all_tests
```

**Your enviroment**
Please run the following and paste the output.
```
$ nix run nixpkgs.nix-info -c nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.3, NixOS, 20.03pre205780.d942688fc13 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(jon): `"home-manager, nixos-20.03pre203904.bb1013511e1"`
 - channels(root): `"nixos-20.03pre205780.d942688fc13"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
pyfakefs 3.7

**Comment by mrbean-bremen** [comment]
Thanks! We didn't notice this due to an older Python 3.7 version used in the Travis tests.
I can confirm that this is reproducible both with Python 3.7.6 and Python 3.8.1, but didn't happen with the previous versions (3.7.5 and 3.8.0). Looks like the `pathlib` implementation has changed in these versions.
I will try to understand that and get a fix together. I will probably make a dot release after this has been fixed.

**Comment by mrbean-bremen** [comment]
For reference - this has been introduced in `pathlib` in [this commit](https://github.com/python/cpython/commit/10ecbadb799ddf3393d1fc80119a3db14724d381).

**Comment by mrbean-bremen** [comment]
Fixed in master and 3.7.1 branch, added [new 3.7.1 PyPi release](https://pypi.org/project/pyfakefs/).
Closing. Please reopen if you still have problems with 3.7.1.
